src/iop/demosaicing/rcd.c

```diff
@@ -684,21 +685,15 @@ static int process_rcd_cl(
     dev_tmp = NULL;
 
     cfa = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(cfa == NULL) goto error;
     VH_dir = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(VH_dir == NULL) goto error;
     PQ_dir = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(PQ_dir == NULL) goto error;
     VP_diff = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(VP_diff == NULL) goto error;
     HQ_diff = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(HQ_diff == NULL) goto error;
     rgb0 = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(rgb0 == NULL) goto error;
     rgb1 = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(rgb1 == NULL) goto error;
     rgb2 = dt_opencl_alloc_device_buffer(devid, sizeof(float) * roi_in->width * roi_in->height);
-    if(rgb2 == NULL) goto error;
+    if(rgb2 == NULL || cfa == NULL || VH_dir == NULL || PQ_dir == NULL || VP_diff == NULL || HQ_diff == NULL || rgb0 == NULL || rgb1 == NULL)
```

Comment:
I find the previous version easier to check and it makes clear that every returned values are checked. Here one has to find the proper variable in the list.

Reply:
I have no strong opinion on this.

A sidenote, as we do the tests here doesn't work anyway because of the lazy allocation. (the first drivers allowed this check, now it always returns a mem_cl != NULL, the allocation is physically done when used! If that goes wrong the kernels returns the error value.

We **could** alloc and check via a test-write a single "data", i tried but that impairs performance in a measurable way.

So - if you prefer, i will revert that part.

Reply:
done